### PR TITLE
Allow `async {}` expressions in const contexts

### PR DESCRIPTION
**Pull request by jonas-schievink**
Gated behind a new `const_async_blocks` feature.

### CODE REVIEW

File: compiler/rustc_feature/src/active.rs [hunk 1]
```diff
@@ -650,6 +650,9 @@ declare_features! (
     /// Allows unsizing coercions in `const fn`.
     (active, const_fn_unsize, "1.53.0", Some(64992), None),
 
+    /// Allows `async {}` expressions in const contexts.
+    (active, const_async_blocks, "1.53.0", None, None),
```
Comment (oli-obk) [inline compiler/rustc_feature/src/active.rs:654]: This should have a tracking issue

Comment (jonas-schievink) [reply compiler/rustc_feature/src/active.rs]: Added https://github.com/rust-lang/rust/issues/85368

**Pull request merged**